bugsnag-android-core/src/main/java/com/bugsnag/android/Error.kt

```diff
@@ -0,0 +1,32 @@
+package com.bugsnag.android
+
+class Error @JvmOverloads internal constructor(
+    var errorClass: String,
+    var errorMessage: String?,
+    var stacktrace: List<Stackframe>,
+    var type: String = "android"
+): JsonStream.Streamable {
+
+    companion object {
+        fun createError(exc: Throwable, projectPackages: Collection<String>): List<Error> {
+            val errors = mutableListOf<Error>()
+
+            var currentEx: Throwable? = exc
+            while (currentEx != null) {
+                val trace = Stacktrace(exc.stackTrace, projectPackages)
+                errors.add(Error(exc.javaClass.name, exc.localizedMessage, trace.trace))
+                currentEx = currentEx.cause
```

Comment:
Do you plan to add a test for adding multiple exceptions in another PR? If not could you ensure the test cover this scenario here?

Reply:
I have added a test for adding multiple exceptions - this uncovered a bug caused by using `exc` rather than `currentExc`, so I've also fixed that.